### PR TITLE
Temporary cookie instead of ServerCookie.

### PR DESCRIPTION
**Pull request by gmaf**
* Working auth without the creation of ServerCookie before redirecting to ironclad.

### CODE REVIEW

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs [hunk 1]
```diff
@@ -174,7 +224,7 @@ public ActionResult Afterlogin(string platform = null, string returnUrl = null)
 
         [HttpPost("~/signin/{platform?}")]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult> Signin(LoginViewModel model, string platform = null)
+        public async Task<IActionResult> Signin(LoginViewModel model, string platform = null)
```
Comment (tarurar) [inline src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: `Aync` postfix

File: src/Lykke.Service.OAuth/CustomCookieAuthenticationEvents.cs [hunk 2]
```diff
@@ -29,17 +31,17 @@ public override async Task ValidatePrincipal(CookieValidatePrincipalContext cont
             }
         }
 
-        public override async Task RedirectToLogin(RedirectContext<CookieAuthenticationOptions> context)
+        public override Task RedirectToLogin(RedirectContext<CookieAuthenticationOptions> context)
```
Comment (tarurar) [inline src/Lykke.Service.OAuth/CustomCookieAuthenticationEvents.cs]: `Async` postfix

File: src/Core/ExternalProvider/LykkeSignInContext.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿using MessagePack;
+
+namespace Core.ExternalProvider
+{
+    /// <summary>
+    ///     Sign in page context.
+    /// </summary>
+    [MessagePackObject(true)]
+    public class LykkeSignInContext
```
[the remark this answers is not in the document]
Comment (gmaf) [reply src/Core/ExternalProvider/LykkeSignInContext.cs]: It's not a message contract, it is a class for saving redirect context.

File: src/Lykke.Service.OAuth.Services/ExternalProvider/ExternalUserOperator.cs [hunk 4]
```diff
@@ -32,7 +25,9 @@ public class ExternalUserOperator : IExternalUserOperator
     {
         private const string RedisPrefixIroncladLykkeLogins = "OAuth:IroncladLykkeLogins";
 
-        private readonly TimeSpan _ironcladLykkeLoginsLifetime = TimeSpan.FromMinutes(2);
+        private readonly TimeSpan _tempLykkeUserIdCookieLifetime = TimeSpan.FromMinutes(2);
```
[the remark this answers is not in the document]
Comment (gmaf) [reply src/Lykke.Service.OAuth.Services/ExternalProvider/ExternalUserOperator.cs]: Right, but later)

File: src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs (hunk 1 above)
Comment (gmaf) [reply src/Lykke.Service.OAuth/Controllers/AuthenticationController.cs]: It is a controller method, no need in Async postfix.

File: src/Lykke.Service.OAuth/CustomCookieAuthenticationEvents.cs (hunk 2 above)
Comment (gmaf) [reply src/Lykke.Service.OAuth/CustomCookieAuthenticationEvents.cs]: It is an implementation of Microsoft class, the method must have an exact name `RedirectToLogin`.

File: src/Core/ExternalProvider/IExternalUserOperator.cs [hunk 5]
```diff
@@ -12,52 +11,94 @@ namespace Core.ExternalProvider
...
+        Task AssociateIroncladUserAsync(LykkeUser lykkeUser, IroncladUser ironcladUser);
+
+        /// <summary>
+        ///     Authenticates lykke user without creating session.
+        /// </summary>
+        /// <param name="username">Username.</param>
+        /// <param name="password">Password.</param>
+        /// <param name="partnerId">Partner id.</param>
+        /// <returns>Lykke user id.</returns>
+        Task<string> AuthenticateLykkeUser(string username, string password, string partnerId);
```
[the remark this answers is not in the document]
Comment (gmaf) [reply src/Core/ExternalProvider/IExternalUserOperator.cs]: Thanks, fixed.

File: src/Lykke.Service.OAuth.Services/ExternalProvider/ExternalUserOperator.cs [hunk 6]
```diff
@@ -76,70 +68,107 @@ public class ExternalUserOperator : IExternalUserOperator
...
+
+        /// <inheritdoc />
+        public void ClearLykkeSignInContext()
+        {
+            _httpContextAccessor.HttpContext.Response.Cookies.Delete(OpenIdConnectConstantsExt.Cookies
+                .LykkeSignInContextCookie);
+        }
+
+        /// <inheritdoc />
+        public async Task<LykkeUser> ProvisionIfNotExistAsync(IroncladUser ironcladUser,
```
[the remark this answers is not in the document]
Comment (gmaf) [reply src/Lykke.Service.OAuth.Services/ExternalProvider/ExternalUserOperator.cs]: Fixed.

**Pull request merged**